parley/src/layout/line/greedy.rs

```diff
@@ -322,34 +344,21 @@ impl<'a, B: Brush> BreakLines<'a, B> {
                                 }
                             }
                             // Otherwise perform an emergency line break
-                            // TODO: make emergency line breaks controllable via a setting
-                            else {
-                                const BREAK_WORDS_IN_EMERGENCY: bool = false;
-                                if BREAK_WORDS_IN_EMERGENCY {
-                                    // If we're at the start of the line, this particular cluster will never fit,
-                                    // so consume it and accept the overflow.
-                                    if self.state.line.x == 0. {
-                                        self.state.append_cluster_to_line(next_x);
-                                        self.state.cluster_idx += 1;
-                                        if is_space {
-                                            self.state.line.num_spaces += 1;
-                                        }
-
-                                        if try_commit_line!(BreakReason::Emergency) {
-                                            self.state.cluster_idx += 1;
-                                            return self.start_new_line();
-                                        }
-                                    } else {
-                                        // Else start a new line
-                                        if try_commit_line!(BreakReason::Emergency) {
-                                            self.state.cluster_idx += 1;
-                                            return self.start_new_line();
-                                        }
-                                    }
-                                } else {
-                                    self.state.append_cluster_to_line(next_x);
-                                    self.state.cluster_idx += 1;
+                            else if let Some(prev_emergency) =
```

Comment:
Is it always correct to try "prev_boundary" first and then fallback to "prev_emergency"? Are there cases (`OverflowWrap::Anywhere`?) where we ought to the later "emergency" breakpoint even if an earlier "regular" breakpoint is available?

I wonder whether we ought to just have one "previous break point" state that just gets set in different conditions depending on the line-breaking styles?

Reply:
> Are there cases (`OverflowWrap::Anywhere`?) where we ought to use the later "emergency" breakpoint even if an earlier "regular" breakpoint is available?

Text with `OverflowWrap::Anywhere` should only use an "emergency" breakpoint if there are no other opportunities to break the line.

The name "anywhere" is a bit confusing since it may give the impression that it behaves like `word-break: break-all`. [but it's just `break-word` with the difference that it affects the min content size](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap#anywhere).

I don't think there's any way to get around having two different previous break points since we need that fallback behavior.

Reply:
> The name "anywhere" is a bit confusing since it may give the impression that it behaves like word-break: break-all

I see. Part of me feels like we ought not to stick to the CSS property names here and come up with something more sensible (but using the CSS properties is also very convenient for Blitz which can translate them 1:1 from Stylo)